omod/src/main/java/org/openmrs/module/fhir2/providers/DiagnosticReportFhirResourceProvider.java

```diff
@@ -69,4 +79,18 @@ public MethodOutcome updateDiagnosticReport(@IdParam IdType id, @ResourceParam D
 		
 		return FhirServerUtils.buildUpdate(service.update(idPart, diagnosticReport));
 	}
+	
+	@Search
+	public Bundle searchForDiagnosticReports(
+	        @OptionalParam(name = DiagnosticReport.SP_ENCOUNTER, chainWhitelist = {
+	                "" }, targetTypes = Encounter.class) ReferenceAndListParam encounterReference,
+	        @OptionalParam(name = DiagnosticReport.SP_PATIENT, chainWhitelist = { "", Patient.SP_IDENTIFIER,
+	                Patient.SP_GIVEN, Patient.SP_FAMILY,
+	                Patient.SP_NAME }, targetTypes = Patient.class) ReferenceAndListParam patientReference,
+	        @OptionalParam(name = DiagnosticReport.SP_ISSUED) DateRangeParam issueDate,
+	        @OptionalParam(name = DiagnosticReport.SP_DATE) DateRangeParam effectiveDate,
```

Comment:
Let's not try to map an effectiveDate. We don't really have anything for which that's semantically meaningful.

Reply:
But we are doing that for observation as well.

Reply:
The meaning is a bit different for a DiagnosticReport versus a standard observation. One of the reasons I want to get them off of this weird obs group formatting.

Reply:
Okay, i'll do that.